lib/fastly_nsq/rake_task.rb

```diff
@@ -34,19 +34,21 @@ def add_rake_task_description_if_one_needed
     end
 
     def run_tasks
+      listener_threads = []
+
       topics.each do |topic|
-        Thread.new do
+        listener_threads << Thread.new do
           wrap_helpful_output(topic) do
             MessageQueue::Listener.new(topic: topic, channel: channel).go
           end
         end
       end
 
-      non_main_threads.map(&:join)
-    end
-
-    def non_main_threads
-      (Thread.list - [Thread.main])
+      while listener_threads.any?(&:status)
+        listener_threads.each do |thread|
+          thread.join(1)
```

Comment:
Double looping here keeps us looping until all threads have finished in some way. Before we blocked on the first thread until it exited... and then on the second after that and so on. This got a process that eventually looped over each thread until all had finished. If we want to allow each one to raise errors (so we can do exception_handling and restarts maybe), we can limit, or we can set `abort_on_exception=true`. The later though this is not guaranteed to raise when you would expect back to the main thread (reading on the internet this weekend gave me this impression) so we would have to know more about the nuances there. You can set this with the `$DEBUG` flag which makes me think it was more for debugging than aborting early?...

This should likely be more verbose (include exception_handling etc.) but I did not want to make to great a change 1: over the weekend, 2: while @jaw6's large PR is still open.